internal/controllers/parser.go

```diff
@@ -295,7 +301,48 @@ func UpdateConfigFromAPIOpts(envoyConfiguration *config.EnvoyConfiguration, prox
 				rt.Action = routeRoute
 
 				routesToAddToVirtualHost = append(routesToAddToVirtualHost, rt)
+			case finalOpts.Auth != nil:
+				fmt.Println("-----")
+				fmt.Printf("finalOpts.Auth not nil - finalOpts.Auth=%+v\n", finalOpts.Auth)
 
+				if finalOpts.Auth.PathPrefix != nil {
+
+				}
+
+				// clusterName := "ext-authz-http-service:9002"
+				upstreamServiceHost := finalOpts.Auth.AuthUpstream.Host.Hostname
+				upstreamServicePort := finalOpts.Auth.AuthUpstream.Host.Port
+
+				// upstreamServiceHost := "ext-authz-http-service"
+				// var upstreamServicePort uint32 = 9002
+				clusterName := generateClusterName(upstreamServiceHost, upstreamServicePort)
+				// clusterName := upstreamServiceHost
+
+				if !envoyConfiguration.ClusterExist(clusterName) {
+					fmt.Printf("cluster does not exist - add: %s %s:%d\n", clusterName, upstreamServiceHost, upstreamServicePort)
+					envoyConfiguration.AddCluster(
+						clusterName,
+						upstreamServiceHost,
+						upstreamServicePort,
+					)
+				} else {
+					fmt.Printf("cluster exists - skip add: %s %s:%d\n", clusterName, upstreamServiceHost, upstreamServicePort)
+				}
+
+				// httpExternalAuthorizationFilter, err := config.NewHTTPExternalAuthorizationFilter("ext-authz-http-service", 9002)
+				httpExternalAuthorizationFilter, err := config.NewHTTPExternalAuthorizationFilter(
+					upstreamServiceHost,
+					upstreamServicePort,
+					clusterName,
+				)
+				httpConnectionManagerBuilder.AppendFilterHTTPExternalAuthorizationFilterToStart(httpExternalAuthorizationFilter)
```

Comment:
I already addressed this above. Is there another way to add ext_auth filter?

Reply:
`HTTPExternalAuthorization` needs to come before `CORS` and `Router` otherwise `envoy` will throw an error.

There is also no point in adding the `HTTPExternalAuthorization` filter if the `auth` section is not defined by the user.

Reply:
Feel free to mark as resolved, if you feel the previous response is enough.